OpenRA.Mods.Common/UpdateRules/Rules/20171014/AddNukeLaunchAnimation.cs

```diff
@@ -0,0 +1,46 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2018 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class AddNukeLaunchAnimation : UpdateRule
+	{
+		public override string Name { get { return "Add 'WithNukeLaunchAnimation' and remove 'NukePower.ActivationSequence'"; } }
+		public override string Description
+		{
+			get
+			{
+				return "The 'ActivationSequence' property has been removed.\n" +
+					"Use the new 'WithNukeLaunchAnimation' trait instead.";
+			}
+		}
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNode actorNode)
+		{
+			foreach (var nuke in actorNode.ChildrenMatching("NukePower"))
+			{
+				var activation = nuke.LastChildMatching("ActivationSequence");
+				if (activation == null)
+					continue;
+
+				var sequence = activation.NodeValue<string>();
+				nuke.RemoveNode(activation);
+				actorNode.AddNode("WithNukeLaunchAnimation", "");
```

Comment:
Same issue as before.  Discovered when running the update rules on D2K:
```
AddNukeLaunchAnimation: Add 'WithNukeLaunchAnimation' and remove 'NukePower.ActivationSequence'
   Updating mod... FAILED

   The automated changes for this rule were not applied because of an error.
   After the issue reported below is resolved you should run the updater
   with SOURCE set to AddNukeLaunchAnimation to retry these changes

   The exception reported was:
     System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
       at System.ThrowHelper.ThrowInvalidOperationException (System.ExceptionResource resource) [0x0000b] in <bb7b695b8c6246b3ac1646577aea7650>:0 
       at System.Collections.Generic.List`1+Enumerator[T].MoveNextRare () [0x00013] in <bb7b695b8c6246b3ac1646577aea7650>:0 
       at System.Collections.Generic.List`1+Enumerator[T].MoveNext () [0x0004a] in <bb7b695b8c6246b3ac1646577aea7650>:0 
       at System.Linq.Enumerable+WhereListIterator`1[TSource].MoveNext () [0x0004e] in <b29fa38c592649ee8d566c045a7d9307>:0 
       at OpenRA.Mods.Common.UpdateRules.Rules.AddNukeLaunchAnimation+<UpdateActorNode>c__Iterator0.MoveNext () [0x000a1] in <300dcefade434fd18783a65f41b5ec39>:0 
       at OpenRA.Mods.Common.UpdateRules.UpdateUtils+<ApplyTopLevelTransform>c__Iterator3.MoveNext () [0x00121] in <300dcefade434fd18783a65f41b5ec39>:0 
       at System.Collections.Generic.List`1[T].InsertRange (System.Int32 index, System.Collections.Generic.IEnumerable`1[T] collection) [0x000ea] in <bb7b695b8c6246b3ac1646577aea7650>:0 
       at System.Collections.Generic.List`1[T].AddRange (System.Collections.Generic.IEnumerable`1[T] collection) [0x00000] in <bb7b695b8c6246b3ac1646577aea7650>:0 
       at OpenRA.Mods.Common.UpdateRules.UpdateUtils.UpdateMod (OpenRA.ModData modData, OpenRA.Mods.Common.UpdateRules.UpdateRule rule, System.Collections.Generic.List`1[System.Tuple`3[OpenRA.FileSystem.IReadWritePackage,System.String,System.Collections.Generic.List`1[OpenRA.MiniYamlNode]]]& files, System.Collections.Generic.HashSet`1[T] externalFilenames) [0x00265] in <300dcefade434fd18783a65f41b5ec39>:0 
       at OpenRA.Mods.Common.UtilityCommands.UpdateModCommand.ApplyRules (OpenRA.ModData modData, System.Collections.Generic.IEnumerable`1[T] rules, System.Boolean skipMaps) [0x0007a] in <300dcefade434fd18783a65f41b5ec39>:0 
```

Reply:
I've checked for all other uses of `AddNode` in this PR and it looks like this is the last case to fix.

Reply:
Fixed.